packages/gatsby/src/utils/webpack-utils.ts

```diff
@@ -219,15 +222,25 @@ export const createWebpackUtils = (
         options: {
           ident: `postcss-${++ident}`,
           sourceMap: !PRODUCTION,
-          plugins: (loader: Loader): Plugin[] => {
+          plugins: (loader: Loader): postcss.Plugin<any>[] => {
+            const autoprefixerOptions = {
+              overrideBrowserslist,
+              flexbox: `no-2009`,
+            }
             plugins =
               (typeof plugins === `function` ? plugins(loader) : plugins) || []
 
-            return [
-              flexbugs,
-              autoprefixer({ overrideBrowserslist, flexbox: `no-2009` }),
-              ...plugins,
-            ]
+            if (plugins && plugins.length) {
+              const autoprefixerOverride: autoprefixer.Autoprefixer = plugins.find(
+                plugin => plugin.postcssPlugin === `autoprefixer`
+              )
+              Object.assign(
+                autoprefixerOptions,
+                autoprefixerOverride.options || {}
+              )
+            }
+
+            return [flexbugs, autoprefixer(autoprefixerOptions), ...plugins]
```

Comment:
I had to look at it a while but I think this can be simplified a bit. For example, something like this:

```suggestion
            const autoPrefixPlugin = autoprefixer({
              overrideBrowserslist,
              flexbox: `no-2009`,
              ...(plugins.find(({postcssPlugin}) => postcssPlugin === `autoprefixer`)?.options ?? {}),
            })

            return [flexbugs, autoPrefixPlugin, ...plugins]
```

(Github doesn't allow me to suggest across the two diff sections but you'd obviously remove the `const autoprefixerOptions` part in favor of this)

You could put the `(plugins.find(({postcssPlugin}) => postcssPlugin === `autoprefixer`)?.options ?? {})` part in a function, like this:
```js
function getPrefixerOptions(plugins){
  const plugin = plugins.find(({postcssPlugin}) => postcssPlugin === `autoprefixer`);
  return plugin?.options ?? {}
}
```